profiling/src/profiling/stack_walking.rs

```diff
@@ -119,19 +127,30 @@ mod detail {
             let hit_rate = stats.hit_rate();
             debug!("Process cumulative {stats:?} hit_rate: {hit_rate}");
         });
+
+        CACHED_STRINGS.with(|cell| {
+            let set: &StringSet = &cell.borrow();
+            // todo: pin down threshold and remove magic number
+            if set.arena_used_bytes() > 4 * 1024 * 1024 {
```

Comment:
Do we have any statistics about how often this would happen? Could we maybe also make this configurable for customers but with a "sane" default?

Reply:
I don't have data yet. That's one thing I'm going to try to discover with some manual testing. For now I'll probably add a `debug!` log line while I test it manually.

Reply:
On a Symfony php-fpm app, I was able to get it up to about 227 KiB. On a WordPress apache app, I couldn't even hit 64 KiB. I did this hammering multiple endpoints with `ab -t 600 -c 2 $url`.

So with two apps, I've seen a pretty big difference in behavior. Hard to say. It'd be nice to have real metrics.